# Resume Scanner & Recommendation

A resume / curriculum vitae scanner machine learning model to recommend the perfect job category based on skills or experience

### Import dependencies

In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings('ignore')
from sklearn.naive_bayes import MultinomialNB
from sklearn.multiclass import OneVsRestClassifier
from sklearn import metrics
from sklearn.metrics import accuracy_score
from pandas.plotting import scatter_matrix
from sklearn.neighbors import KNeighborsClassifier
from sklearn import metrics

### Import dataset

In [29]:
resumeDataSet = pd.read_csv('./UpdatedResumeDataSet.csv', encoding='utf-8')
print(resumeDataSet.describe(include='all'))
print('\n')
print(resumeDataSet.head())
print('\n')
print(sorted(resumeDataSet['Category'].unique()))

              Category                                             Resume
count              962                                                962
unique              25                                                166
top     Java Developer  Technical Skills Web Technologies: Angular JS,...
freq                84                                                 18


       Category                                             Resume
0  Data Science  Skills * Programming Languages: Python (pandas...
1  Data Science  Education Details \r\nMay 2013 to May 2017 B.E...
2  Data Science  Areas of Interest Deep Learning, Control Syste...
3  Data Science  Skills • R • Python • SAP HANA • Table...
4  Data Science  Education Details \r\n MCA   YMCAUST,  Faridab...


['Advocate', 'Arts', 'Automation Testing', 'Blockchain', 'Business Analyst', 'Civil Engineer', 'Data Science', 'Database', 'DevOps Engineer', 'DotNet Developer', 'ETL Developer', 'Electrical Engineering', 'HR', 'Hadoop', 'He

In [24]:
resumeDataSet['Category'].sort_values().value_counts()

Category
Java Developer               84
Testing                      70
DevOps Engineer              55
Python Developer             48
Web Designing                45
HR                           44
Hadoop                       42
Data Science                 40
Operations Manager           40
Sales                        40
Blockchain                   40
ETL Developer                40
Mechanical Engineer          40
Arts                         36
Database                     33
Electrical Engineering       30
Health and fitness           30
PMO                          30
Business Analyst             28
DotNet Developer             28
Automation Testing           26
Network Security Engineer    25
SAP Developer                24
Civil Engineer               24
Advocate                     20
Name: count, dtype: int64

#### Duplicate data

In [42]:
print(resumeDataSet.duplicated().value_counts())
print('Duplicate Data Ratio')
duplicateDataRatio = resumeDataSet.duplicated().value_counts()[False]/resumeDataSet.duplicated().value_counts()[True] * 100
print(str(round(duplicateDataRatio, 2)) + "%")

True     796
False    166
Name: count, dtype: int64
Duplicate Data Ratio
20.85%


From the dataset, 20.85% of the data are unique, meaning that the rest 79.15% of the data are duplicates

#### Missing data

In [31]:
resumeDataSet.notnull().value_counts()

Category  Resume
True      True      962
Name: count, dtype: int64

The dataset contains 0 missing values